File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/javadoc/comment/direct/Java/C/s831281137_commented.c
#include <stdio.h>

#define MAX_N 100
#define MAX_W 100

int v[MAX_N + 1];
int w[MAX_N + 1];
int sumV[MAX_N + 1][MAX_W + 1];

int nint() {
    int x;
    scanf("%d", &x);
    return x;
}

int main() {
    int N = nint();
    int W = nint();

    for (int i = 1; i <= N; i++) {
        v[i] = nint();
        w[i] = nint();
    }

    for (int i = 1; i <= N; i++) {
        for (int i_sum = 0; i_sum < w[i]; i_sum++) {
            sumV[i][i_sum] = sumV[i - 1][i_sum];
        }

        for (int i_sum = w[i]; i_sum <= W; i_sum++) {
            sumV[i][i_sum] = (sumV[i - 1][i_sum] > sumV[i - 1][i_sum - w[i]] + v[i]) ? sumV[i - 1][i_sum] : sumV[i - 1][i_sum - w[i]] + v[i];
        }
    }

    printf("%d\n", sumV[N][W]);

    return 0;
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s831281137_commented.py
import sys

def main():
    input = sys.stdin.read
    data = input().split()
    index = 0

    def nint():
        nonlocal index
        index += 1
        return int(data[index - 1])

    N = nint()
    W = nint()

    v = [0] * (N + 1)
    w = [0] * (N + 1)

    for i in range(1, N + 1):
        v[i] = nint()
        w[i] = nint()

    sumV = [[0] * (W + 1) for _ in range(N + 1)]

    for i in range(1, N + 1):
        for i_sum in range(w[i]):
            sumV[i][i_sum] = sumV[i - 1][i_sum]

        for i_sum in range(w[i], W + 1):
            sumV[i][i_sum] = max(sumV[i - 1][i_sum], sumV[i - 1][i_sum - w[i]] + v[i])

    print(sumV[N][W])

if __name__ == "__main__":
    main()  # End of Code
